### ISSUE TITLE
Support GraphQL variables

### ISSUE THREAD

**Issue by spenserblack** [XAMPPRocky/octocrab#177]
If the body is, for example
```graphql
query ($number_of_repos: Int!) {
  viewer {
    name
    repositories(last: $number_of_repos) {
      nodes {
        name
      }
    }
  }
}
```
Then there should be a simple way to pass variables like
```json
{"number_of_repos": 3}
```

While this may be a breaking change, I think it would be best if `Octocrab::graphql` returned a handler, similar to `Octocrab::issues`.
Then the api could be something like
```rust
let result: serde_json::Value = octocrab::instance()
    .graphql(query)
    .variable("number_of_repos", 3)
    .send()
    .await?;
```

***

See [these docs](https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#working-with-variables) for an example.

**Comment by XAMPPRocky** [comment]
Thank you for your issue! I think this would probably be better left to a fully fledged graphql library. Octocrab should be compatible with any in the ecosystem.

**Comment by spenserblack** [comment]
>I think this would probably be better left to a fully fledged graphql library.

That makes sense, but if GraphQL doesn't have support in octocrab should it be marked as deprecated in favor of such a library?

>Octocrab should be compatible with any in the ecosystem.

I'm not sure what you mean by this. Could you elaborate? Thanks!

**Comment by XAMPPRocky** [comment]
> That makes sense, but if GraphQL doesn't have support in octocrab should it be marked as deprecated in favor of such a library?

Sorry I might not have been clear. There are graphql libaries like [`graphql-client`](https://github.com/graphql-rust/graphql-client), which allow you to build a query, and that will generate a JSON object that is `serde::Serialize`, you can then pass that object to `octocrab`'s `graphql` endpoint. Does that make sense?

**Comment by spenserblack** [comment]
Yes, that makes sense, but I'm not sure if it will work. I'll have to play with it, but, looking at the source code, I think passing the variables to the `graphql` endpoint would create a malformed body.
https://github.com/XAMPPRocky/octocrab/blob/9eb052ca5d6cb51653effb6fb8c41eae3620f9eb/src/lib.rs#L588-L600

By passing the body from `graphql-client` to `Octocrab::graphql`, wouldn't that put the variables in the the body of the *query* value?
If you look at [the example in `graphql-client`'s documentation](https://docs.rs/graphql_client/0.10.0/graphql_client/trait.GraphQLQuery.html), the variables are *outside* the `query` value.

Here's a gist with a working example using `Octocrab::post`.
https://gist.github.com/spenserblack/f90dd2c5dc652015d5bfc8f2c6f2b63d
